src/path.cc

```diff
@@ -268,4 +267,42 @@ std::string PathResolve(Environment* env,
 }
 #endif  // _WIN32
 
+#ifdef _WIN32
+constexpr static bool IsWindowsDeviceRoot(const char code) noexcept {
+  return (code >= 'a' && code <= 'z') || (code >= 'A' && code <= 'Z');
+}
+#endif  // _WIN32
+
+void ToNamespacedPath(Environment* env, BufferValue* path) {
+#ifdef _WIN32
+  if (path->length() == 0) return;
+  auto resolved_path = node::PathResolve(env, {path->ToStringView()});
+
+  if (resolved_path.size() <= 2) {
+    path->SetLength(resolved_path.size());
+    memcpy(&path, resolved_path.c_str(), resolved_path.size());
+    return;
+  }
+
+  if (resolved_path[0] == '\\') {
+    // Possible UNC root
+    if (resolved_path[1] == '\\') {
+      if (resolved_path[2] != '?' && resolved_path[2] != '.') {
```

Comment:
Here you know that resolved_path has size 3 or more, so this is safe. You *may* want to add a comment (this is not a request for change, just a remark).

Reply:
Definitely, agreed. Adding a comment to here.